corehq/apps/domain/forms.py

```diff
@@ -1370,6 +1371,14 @@ def save(self, commit=True):
                 if not account_save_success:
                     return False
 
+                # changing a plan overrides future subscriptions
+                future_subscriptions = Subscription.objects.filter(
+                    subscriber=self.domain,
+                    date_start__gte=datetime.date.today()
```

Comment:
one last change - this should be ```date_start__gt``` because otherwise we'll include the current subscription and its ```date_end``` should be set to today (handled by ```change_plan```) rather than its ```date_start```.